portal/README.md

```diff
@@ -1,32 +1,108 @@
 # Portal
 
+# Prerequisite
+
+1. Need to start `db` and `redis` container
+2. Need to start authgear servers
+
 # Setup locally
 
-We need to setup the environment variable `AUTHGEAR_CLIENT_ID=portal` and `AUTHGEAR_ENDPOINT=http://localhost:3000`.
+## Setup proxy
+
+We need a proxy server for us to call different server with the same domain / port. In `/portal` directory, there is a nginx container for proxy.
+
+To start proxy container
+
+```
+# in /portal
+docker-compose up -d
+```
+
+This container listen to port 8000. For redirection configuration, please refer to nginx config `/portal/nginx.conf`
+
+## Setup environment variable
+
+We need to setup the environment variable `AUTHGEAR_CLIENT_ID=portal` and `AUTHGEAR_ENDPOINT=http://localhost:3000`, and start graphQL server.
+
+## Setup graphQL server
+
+```sh
+AUTHGEAR_CLIENT_ID=portal AUTHGEAR_ENDPOINT=http://localhost:3000 go run ./cmd/portal start
+```
+
+## Setup portal development server
+
+1. Install dependencies
+
+```
+npm install
+```
+
+2. Run development server
+
+```
+npm run start
+```
+
+This command should start a web development server on port 1234.
+
+3. Configure authgear.yaml
 
 We need the following `authgear.yaml` to setup authgear for the portal.
 
 ```yaml
 http:
   allowed_origins:
-  # The SDK uses XHR to fetch the OAuth/OIDC configuration,
-  # So we have to allow the origin of the portal.
-  - localhost:8000
+    # The SDK uses XHR to fetch the OAuth/OIDC configuration,
+    # So we have to allow the origin of the portal.
+    - localhost:8000
 oauth:
   clients:
-  # Create a client for the portal.
-  # Since we assume the cookie is shared, there is no grant nor response.
-  - client_id: portal
-    # Note that the trailing slash is very important here
-    # URIs are compared byte by byte.
-    redirect_uris:
-    - "http://localhost:8000/"
-    post_logout_redirect_uris:
-    - "http://localhost:8000/"
-    grant_types: []
-    response_types: ["none"]
+    # Create a client for the portal.
+    # Since we assume the cookie is shared, there is no grant nor response.
+    - client_id: portal
+      # Note that the trailing slash is very important here
+      # URIs are compared byte by byte.
+      redirect_uris:
+        - "http://localhost:8000/"
+      post_logout_redirect_uris:
+        - "http://localhost:8000/"
+      grant_types: []
+      response_types: ["none"]
+```
+
+## Sign-up
+
+When you try to access the portal through proxy `http://localhost:8000` (port 8000), you will be redirected to authgear sign up / sign in page in port 3000 (hosted by authgear main server)
+
+NOTE: make sure the authgear server is in development mode, which skips sending actual email
+
+Follow the instruction on the webpage,
+
+1. Click `Create One` link
+2. Input an email address
+3. The website will ask for verification code, to get verified, inspect the log from authgear main server
+
+```
```

Comment:
We don't need to include these logs in the README. Instead, we can instruct the developer to find a WARN level log entry, which should have distinct color on console.